traits/tests/test_tuple.py

```diff
@@ -73,3 +73,22 @@ class A(HasTraits):
         b = A()
         self.assertEqual(a.foo, (0, ("", 0)))
         self.assertIs(a.foo, b.foo)
+
+    def test_lists_not_accepted(self):
```

Comment:
This is just an additional test for existing behaviour that isn't changed by this PR.

Reply:
Is it worth adding the same tests for `BaseTuple` as well?

Reply:
Not just yet - I'm deliberately avoiding changing the behaviour of `BaseTuple` at all in this PR (at least partly because we have other `BaseTuple` subclasses). I do plan a followup PR at some point for `BaseTuple`.